editor/utils/block-list.js

```diff
@@ -40,10 +59,19 @@ const INNER_BLOCK_LIST_CACHE = {};
 export function createInnerBlockList( uid, renderBlockMenu = noop ) {
 	if ( ! INNER_BLOCK_LIST_CACHE[ uid ] ) {
 		const InnerBlockListComponent = class extends Component {
-			componentWillReceiveProps( nextProps ) {
-				this.updateNestedSettings( {
-					supportedBlocks: nextProps.allowedBlocks,
-				} );
+			constructor( props ) {
+				super( props );
+				this.state = {};
+			}
+
+			static getDerivedStateFromProps( props ) {
+				const newSettings = {
+					allowedBlocks: props.allowedBlocks,
+					locking: getHighestLocking( props.locking, props.parentLocking ),
```

Comment:
Why do we compare the `locking` prop with the parent `locking` prop? Shouldn't we just always use the prop passed to `InnerBlocks`?

Reply:
Hi @youknowriad, this is what allows us to propagate the locking to descendants. Even if a child block sets no locking at all or a less restrictive locking the locking of the parent applies. So for example, if the parent adds locking and uses columns block the locking applies also to columns. I'm not certain if this is the right behavior and in fact, I'm now considering changing to just propagate the locking if the child block did not set locking at all.

Reply:
yes, I'd consider `props.locking || props.parentLocking` a better approach here :) (or something in that vein. We could also question whether propagation is something we want (I know we do this at the moment)